### ISSUE TITLE
New cop for redundant braces for hash as a last argument

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#14592]
Sorry if this exists somewhere, but I was not able to find it.

If hash is a last argument, no need to wrap it by explicitly by braces:

```ruby
# bad
foo({ x: 1, y: 2 })
foo(1, { x: 1 })

# good
foo(x: 1, y: 2)
foo(1, x: 1)
foo({ x: 1 }, { y: 2 })
```

**Comment by Earlopain** [comment]
I feel like I would prefer the explicit braces. If the method signature is `def foo(bar, baz)`, passing keywords feels weird to me.

There is also a chance for false positives if the method signature contains `**nil` (explicitly not accepting keyword arguments). Also, it changes behaviour in the following case:

```rb
def foo(b, **kwargs) = puts kwargs

foo({ x: 1, y: 2 }) # => {}
foo(x: 1, y: 2) # => wrong number of arguments (given 1, expected 2)

def bar(a = nil, **kwargs) = puts kwargs

bar({ x: 1, y: 2 }) # => {}
bar(x: 1, y: 2) # => {x: 1, y: 2}
```

**Comment by koic** [comment]
Yeah, the `Style/BracesAroundHashParameters` cop once existed, but it was removed due to an incompatible change in Ruby 3.0. There is no way to resolve this in RuboCop's static analysis without false positives. For more details, see #7643.
